timeseries/src/autogluon/timeseries/models/multi_window/multi_window_model.py

```diff
@@ -213,12 +214,25 @@ def score_and_cache_oof(
         store_predict_time: bool = False,
         **predict_kwargs,
     ) -> None:
-        # self.val_score, self.predict_time, self._oof_predictions already saved during _fit()
```

Comment:
I had a look at the rest of the code but still not sure why we need to modify the `score_and_cache_oof` method. What is the new case when we need to pass `val_data` to `MultiWindowBacktestingModel.score_and_cache_oof`? 

As I understand, previously there were two branches:
1. Custom `val_data` is provided by the user, `num_val_windows=0` -> in this case `MultiWindowBacktestingModel` is not used.
2. User only provides `train_data` and `num_val_windows > 0` -> in this case `MultiWindowBacktestingModel` is used and it stores the `val_score` during `fit`; `score_and_cache_oof` is a no-op.